docs/rfc/rfc-2.adoc

```diff
@@ -0,0 +1,220 @@
+  `lotSizeSatoshis` view method.
+- The previous outpoint describing the deposit UTXO, available on the tBTC
+  deposit contract from the `fundingInfo` view method, which returns it as
+  the third `utxoOutpoint` entry in its 3-part return tuple.
+- The keep's public key, available on the Keep contract from the
+  `getPublicKey` view method.
+- The deposit redemption script, straightforwardly computable from the keep's
+  public key in a way detailed in the section on <<output-script>>.
+- One Bitcoin address per signer, specified on the client in a way detailed
+  in the section on <<recovery-address>>.
+- The keep's signature on the sighash produced by the previous items,
+  computed via the threshold ECDSA protocol in a way detailed in the section on
+  <<signing>>.
+- F, the desired fee, which is a function of the going feerate on the Bitcoin
+  network and the number of bytes in the final transaction.
+
+The desired fee is the most problematic one to procure under the goal of not
+strictly requiring a Bitcoin node connection, as generally a Bitcoin node
+provides estimates of the feerates that will achieve transaction confirmation
+within a certain amount of time. A short-term solution is proposed in the
+section on <<bitcoin-fee>>.
+
+[[output-script]]
+===== Computing the Deposit Output Script
+
+[[recovery-address]]
+===== Specifying an Operator's Bitcoin Recovery Address
+
+To specify a given operator's Bitcoin Recovery Address, this proposal suggests
+adding a single key to the `Extensions.TBTC` config entry, `BTCRefundTarget`.
+The value for this key should be one of:
+
+- A simple BTC address.
+- An xpub, ypub, or zpub that allows the client to derive new addresses for
+  each refund.
+
+The challenge with a *pub is that, to avoid reuse, it needs to do one of two
+things:
+
+- Be able to check, on-chain, whether a given address has been used; or,
+- Have a *pub that is only meant for use by the client, and have the client
+  track the latest address generated (as an index starting at 0).
+
+To avoid _strictly requiring_ a connection to a Bitcoin node, this RFC proposes
+tracking the latest address generated, and allowing an additional config key,
+`BTCRefundTargetDerivationStartIndex`, that sets a starting point. The client
+should write the updated index to the `current/` directory that tracks
+active key shares, to an encrypted file named
+`current-btc-refund-derivation-index`. The operator should then seek to obtain
+from the staker an extended public key that is only used for this purpose.
+
+If an optional Bitcoin node connection is available, the client should
+additionally check a derived address for a balance before publishing. A fallback
+could be implemented that checks
+`https://blockstream.info/api/address/<address>` for a given derived address to
+ensure `tx_count` is 0.
+
+[[bitcoin-fee]]
+===== Dealing with the Bitcoin Fee
+
+// Each client sets a max fee per vbyte in config. Signing exchanges each
```

Comment:
Why some sections are commented out?

Reply:
These are the “I need to fill in more detail, but here are my high-level thoughts” bits. Also why this is still a draft 😅

I'm ~80% of the way through filling in a lot of these, should be able to push later this morning.

Reply:
These are now done.